Avoid the Xorg build-time dependency, as we already know the answers
to the questions that are trying to be determined.

diff --git a/meson.build b/meson.build
index bf0e58506..7d8f964c3 100644
--- a/meson.build
+++ b/meson.build
@@ -82,8 +82,9 @@ endif

 # X11
 # defaults
-x_path = '/usr/bin/X11:/usr/X11R6/bin:/opt/X11R6/bin'
-x_bin = '/usr/bin/X'
+x_path = '/usr/bin'
+x_bin = '/usr/bin/Xorg'
+xserver_has_listen = true
 if have_x11_support
   xcb_dep = dependency('xcb')
   x_deps = declare_dependency(
@@ -92,24 +93,6 @@ if have_x11_support
       dependency('xau'),
     ],
   )
-  # Xserver 1.17 & later default to -nolisten and require -listen for remote access
-  xserver_deps = dependency('xorg-server', version : '>=1.17', required : false)
-  xserver_has_listen = xserver_deps.found()
-  find_x_server_script = find_program('build-aux/find-x-server.sh', native: true)
-  find_x_server_out = run_command(find_x_server_script).stdout().strip()
-  if find_x_server_out != ''
-    x_bin = find_x_server_out
-    x_bin_path_split = x_bin.split('/')
-    i = 0
-    x_path = '/'
-    foreach dir : x_bin_path_split
-      if i < x_bin_path_split.length() - 1
-        x_path = x_path / dir
-      endif
-      i = i + 1
-    endforeach
-  endif
-
   xdmcp_dep = cc.find_library('Xdmcp', required: get_option('xdmcp'))
   have_xdmcp = xdmcp_dep.found()
   if have_xdmcp
